### PR TITLE
Throw errors on failed `fetch`

### LINKED ISSUE #1226: Confusing error message when snap files return status 404
The error message thrown when snap files return 404 is related to manifest validation and not fetching. This seems flawed.

### PR DESCRIPTION
**Pull request by FrederikBolding**
Throw errors if `fetch` fails when trying to fetch snaps, for example if a snap URL results in 404. Previously these functions would return garbage data by trying to parse 404 responses.

Fixes #1226

### CODE REVIEW

File: packages/snaps-controllers/src/snaps/location/http.ts [hunk 1]
```diff
@@ -56,9 +56,11 @@ export class HttpLocation implements SnapLocation {
       this.url,
     ).toString();
 
-    const contents = await (
-      await this.fetchFn(canonicalPath, this.fetchOptions)
-    ).text();
+    const response = await this.fetchFn(canonicalPath, this.fetchOptions);
+    if (!response.ok) {
+      throw new Error(`Failed to fetch "${canonicalPath}".`);
```
Comment (Mrtenz) [inline packages/snaps-controllers/src/snaps/location/http.ts:61]: Could be helpful to include a status code here:
```suggestion
      throw new Error(`Failed to fetch "${canonicalPath}". Status code: ${response.status}.`);
```

File: packages/snaps-controllers/src/snaps/location/http.ts [hunk 2]
```diff
@@ -84,6 +86,9 @@ export class HttpLocation implements SnapLocation {
 
     const canonicalPath = this.toCanonical(relativePath).toString();
     const response = await this.fetchFn(canonicalPath, this.fetchOptions);
+    if (!response.ok) {
+      throw new Error(`Failed to fetch "${canonicalPath}".`);
```
Comment (Mrtenz) [inline packages/snaps-controllers/src/snaps/location/http.ts:90]: Same here.

File: packages/snaps-controllers/src/snaps/location/npm.ts [hunk 3]
```diff
@@ -222,9 +222,13 @@ async function fetchNpmTarball(
...
   fetchFunction: typeof fetch,
 ): Promise<[ReadableStream, SemVerVersion]> {
-  const packageMetadata = await (
-    await fetchFunction(new URL(packageName, registryUrl).toString())
-  ).json();
+  const packageResponse = await fetchFunction(
+    new URL(packageName, registryUrl).toString(),
+  );
+  if (!packageResponse.ok) {
+    throw new Error('Failed to fetch NPM registry entry');
```
Comment (Mrtenz) [inline packages/snaps-controllers/src/snaps/location/npm.ts:229]:
```suggestion
    throw new Error(`Failed to fetch NPM registry entry. Status code: ${response.status}.`);
```

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: packages/snaps-controllers/src/snaps/location/http.ts (hunk 2 above)
Comment (FrederikBolding) [inline packages/snaps-controllers/src/snaps/location/http.ts:90]:
```suggestion
      throw new Error(`Failed to fetch "${canonicalPath}". Status code: ${response.status}.`);
```

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**